io.embrace.sdk/Scripts/Native/Embrace_iOS.cs

```diff
@@ -12,246 +12,264 @@ namespace EmbraceSDK.Internal
     ///  Embrace_iOS uses statically linked methods that are linked into the executable using the DllImportattribute.  For iOS, we have to use “__Internal” as the library name for the attribute [DllImport("__Internal")].  
     ///  When a user makes a call to our Embrace class this is passed on to a method from the provider interface which then calls the external method. 
     /// </summary>
-#if UNITY_IOS || UNITY_TVOS
+#if (UNITY_IOS || UNITY_TVOS)
     [ExcludeFromCoverage]
     public class Embrace_iOS : IEmbraceProvider
     {
+        // The first prime greater than 255. The number of views in the dictionary managed from Unity's side is *probably* going to be 255 or less.
+        // So choosing the first prime greater than that number to reduce the chance of collisions (and therefore resizing).
+        private const int INITIALCAPACITY = 257;
+        
+        // The ConcurrencyLevel (first argument) is recommended by the example docs for ConcurrentDictionary.
+        ConcurrentDictionary<string, string> _viewDictionary = new ConcurrentDictionary<string, string>(Environment.ProcessorCount * 2, INITIALCAPACITY);
+        
         [DllImport("__Internal")]
-        private static extern void embrace_sdk_start();
+        private static extern bool embrace_sdk_start_native(string appId, string appGroupId, string baseUrl,
+            string devBaseUrl, string configBaseUrl);
 
         [DllImport("__Internal")]
-        private static extern void embrace_sdk_endAppStartup(string properties);
+        private static extern bool embrace_sdk_is_started();
 
         [DllImport("__Internal")]
-        private static extern int embrace_sdk_getLastRunEndState();
-
+        private static extern void crash();
+        
         [DllImport("__Internal")]
-        private static extern void embrace_sdk_setUserIdentifier(string identifier);
-
+        private static extern void set_unity_metadata(string unityVersion, string guid, string sdkVersion);
+        
         [DllImport("__Internal")]
-        private static extern void embrace_sdk_clearUserIdentifier();
+        private static extern void end_session();
 
         [DllImport("__Internal")]
-        private static extern void embrace_sdk_setUsername(string username);
+        private static extern IntPtr get_device_id();
 
         [DllImport("__Internal")]
-        private static extern void embrace_sdk_clearUsername();
+        private static extern IntPtr get_session_id();
```

Comment:
Is there any risk of symbol clashing now that these functions don't have an `embrace` prefix?

Reply:
There probably is actually. I hadn't thought about that. Thank you for that catch. I'll edit the Swift and embrace code to add the prefix.